src/manifest.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+const Address = require('./address')
+const { Block } = require('multformats/block')
+const { sha256: hasher } = require('multiformats/hashes/sha2')
+const mhtype = 'sha2-256'
+const codec = require('@ipld/dag-cbor')
```

Comment:
the idea is to move this to io and still interact with the Block data type through some encode/decode methods but it take care of the hasher, codec, and subsequent writing to ipfs.